src/open_samus_returns_rando/files/templates/randomizerpowerup.lua

```diff
@@ -96,6 +97,34 @@ function RandomizerPowerup.MarkLocationCollected(actorOrName)
     Blackboard.SetProp(playerSection, propName, "b", true)
 end
 
+local AeionAbilityClouds = {
+    {"s000_surface", "LE_SpecialAbility_ScanningPulse"},
+    {"s028_area2c", "LE_SpecialAbility_EnergyShield"},
+    {"s030_area3", "LE_SpecialAbility_EnergyWave"},
+    {"s060_area6", "LE_SpecialAbility_PhaseDisplacement"},
+}
+
+function RandomizerPowerup.ActivateSpecialEnergy(actorOrName)
+    local name = actorOrName.sName
+    local cloud = "TG_SpecialEnergyCloud"
+    local trigger = cloud .. string.sub(name, 11)
```

Comment:
This will throw a lua exception for metroids. `actorOrName` is either an actor or really just a name a.k.a. a string. In the later case `sName` doesn't exist on it, so `local name` is `nil` and you cannot use `nil` with `string.sub`.

For reference see the check in `MarkLocationCollected`. Probably worth refacotring the check out and do it before calling both methods such that both only get a string.

Reply:
Couldn't I say `name = actorOraname.sName if name ~= nil`?

Reply:
Ah I see the ch3ck in that function. I'll pull that out